scipy/stats/_binned_statistic.py

```diff
@@ -570,10 +570,13 @@ def binned_statistic_dd(sample, values, statistic='mean',
         result.fill(0)
         flatcount = np.bincount(binnumbers, None)
         a = flatcount.nonzero()
-        for i in np.unique(binnumbers):
-            for vv in builtins.range(Vdim):
+        for vv in builtins.range(Vdim):
+            for i in np.unique(binnumbers):
                 # NOTE: take std dev by bin, np.std() is 2-pass and stable
-                result[vv, i] = np.std(values[vv, binnumbers == i])
+                binned_data = values[vv, binnumbers == i]
+                # calc std only when binned data is 2 or more for speed up.
+                if len(binned_data) >= 2:
+                    result[vv, i] = np.std(binned_data)
```

Comment:
what about the cases when `binned_data < 2` ? currently `numpy.std` returns a `0.0` and this would omit those values entirely and would not be backwards compatible with 1.4.x versions of SciPy.

Reply:
Instead you could add a test case for `std` routine with a binned_data where only 1 entry was in the bin and have that bin calculate a `0.0` value. We don't have one of those currently AFAIK.

Reply:
In my understanding, all `result` items are initialized with 0.  `numpy.std` returns `0.0` when the number of binned_data is less than 2 (0 or 1). In that cases, we don't need to calculate `np.std`, that makes calculation faster. But my understanding might be wrong...

>  this would omit those values entirely
Could you explain the details of this?

Reply:
>In my understanding, all result items are initialized with 0.

Looking at it again today, you're right. I must've been looking at the `'mean'` case right above it where they default to `nan`s. 

The case when the number of binned data are 0 is perhaps a little less obvious that those would be `0`. I think that should be mentioned explicitly in the docstring. 

I don't think it was possible to do that before we added the functionality to re-use a binned statistic calculation which is a more recent change.

Reply:
> @rlucas7 First of all, thank you so much for taking your time. I'm sorry I was not aware that this PR break backward compatibility. If so, I would like to remove the change. Before that, please take a look my comments in the reviews.

I left a comment to clarify.

Reply:
> I think that should be mentioned explicitly in the docstring.
I did it.

Reply:
Yes, Thanks for that. Marking resolved.